pkg/volume/local/local.go

```diff
@@ -281,6 +282,25 @@ func (plugin *localVolumePlugin) getGlobalLocalPath(spec *volume.Spec) (string,
 	}
 }
 
+func (plugin *localVolumePlugin) RequiresFSResize() bool {
+	return true
+}
+
+func (plugin *localVolumePlugin) NodeExpand(resizeOptions volume.NodeResizeOptions) (bool, error) {
+	fsVolume, err := util.CheckVolumeModeFilesystem(resizeOptions.VolumeSpec)
+	if err != nil {
+		return false, fmt.Errorf("error checking VolumeMode: %v", err)
+	}
+	if !fsVolume {
+		return true, nil
+	}
+	_, err = util.GenericResizeFS(plugin.host, plugin.GetPluginName(), resizeOptions.DevicePath, resizeOptions.DeviceMountPath)
```

Comment:
What happens if the local volume is a directory in a shared filesystem instead of a dedicated disk? Do we end up expanding the shared filesystem and is that expected?

Reply:
yes, i have analysed this, because local volume don't limit logic type(eg. lvm or other technology), for disk type ,it's not easy to look up suitable tools do the same thing as fs type. so i think this commit just a workaround for #101340

Reply:
i havn't thought about this , GenericResizeFS only expand ext3/ext4/btrfs/xfs, for shared file system like ntfs, no operation.

Reply:
@zhanghe9702 I am sorry, I did not see an update to this PR and hence I have gone ahead and implemented NO-OP for directory based local volumes in https://github.com/kubernetes/kubernetes/pull/102886 . But if you want to update this PR, I will go ahead and close mine.